### PR TITLE
John/backend

### PR DESCRIPTION
**Pull request by johnwroge**
Created cookieController and sessionController with methods startSession and setSSIDCookie and added these to router.post('/login). Also created Session mongoose schema.

### CODE REVIEW

File: server/controllers/cookieController.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const cookieController = {};
+
+cookieController.setCookie = (req, res, next) => {
```
Comment (atu816) [inline server/controllers/cookieController.js]: Is this being used anywhere?

File: server/controllers/dbController.js [hunk 2]
```diff
@@ -91,11 +91,13 @@ dbController.verifyUser = (req, res, next) => {
 
   // Pull out the username and password from the request body
   const { username, password } = req.body;
+  
   console.log("Received username: " + username + " and password: " + password);
 
   // Find the user in the database
   User.findOne({ username: username })
     .then((user) => {
+      console.log(user)
```
Comment (atu816) [inline server/controllers/dbController.js]: This gets logged below inside the console.group() / console.groupEnd()

File: server/controllers/sessionController.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+const sessionController = {};
+
+sessionController.isLoggedIn = (req, res, next) => {
+    console.log("\n");
+    // Log to let us know we're in the controller
+    console.log("\u001b[1;32m sessionController.isLoggedIn called");
+
+    Session.findOne({ cookieId: req.cookies.ssid }, (err, session) => {
+        if (err) {
+            return next('Error in sessionController.isLoggedIn: ' + JSON.stringify(err));
```
Comment (atu816) [inline server/controllers/sessionController.js]: Look at some other error catches and try to make your next(____) consistent with the other error logs that get returned.

**Review by atu816 — CHANGES_REQUESTED** [review]
- Error logs are inconsistent with other error catches.
- Double check if you need both cookie middleware functions you set up in the controller?
- Minor comments otherwise

**Review by jbhowat — APPROVED** [review]

File: server/controllers/sessionController.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+sessionController.startSession = (req, res, next) => {
+    console.log("\n");
+  // Log to let us know we're in the controller
+  console.log("\u001b[1;32m sessionController.startSession called ");
+  
+    // console.log('res.locals', res.locals.user.id)
+   
+    // res.locals.user_info = user;
+    Session.create({ cookieId : res.locals.user_info.user_id }, (err, session) => {
+        if (err) return next('Error in sessionController.startSession: ' + JSON.stringify(err));
```
Comment (atu816) [inline server/controllers/sessionController.js:36]: Same error log thing here. Going to put it through but edit it for next PR.

**Review by atu816 — APPROVED** [review]
One more error consistency to get in the next PR but approving

**Pull request closed without merging**